File: service-proxy-examples/service-consumer/src/main/java/io/vertx/examples/service/consumer/ConsumerVerticle.java
package io.vertx.examples.service.consumer;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.json.JsonObject;
import io.vertx.examples.service.ProcessorService;
import io.vertx.launcher.application.VertxApplication;


/**
 * A verticle consuming the provided {@link ProcessorService} service.
 */
public class ConsumerVerticle extends AbstractVerticle {

  public static void main(String[] args) {
    VertxApplication.main(new String[]{ConsumerVerticle.class.getName(), "-cluster"});
  }

  @Override
  public void start() {
    ProcessorService service = ProcessorService.createProxy(vertx, "vertx.processor");

    JsonObject document = new JsonObject().put("name", "vertx");

    service.process(document)
      .onSuccess(json -> {
      System.out.println(json.encodePrettily());
    }).onFailure(failure -> {
      System.out.println(failure);
      Failures.dealWithFailure(failure.getCause());
    });
  }
}

File: service-proxy-examples/service-consumer/src/main/kotlin/io/vertx/examples/service/consumer/ConsumerVerticle.kt
package io.vertx.examples.service.consumer

import io.vertx.examples.service.ProcessorService
import io.vertx.kotlin.core.json.*

class ConsumerVerticle : io.vertx.core.AbstractVerticle()  {
  override fun start() {
    var service = ProcessorService.createProxy(vertx, "vertx.processor")

    var document = json {
      obj("name" to "vertx")
    }

    service.process(document, { r ->
      if (r.succeeded()) {
        println(r.result().toString())
      } else {
        println(r.cause())
        io.vertx.examples.service.consumer.Failures.dealWithFailure(r.cause())
      }
    })
  }
}
